internal/cluster/kubernetes/user_namespace_deleter.go

```diff
@@ -55,6 +56,17 @@ func (d UserNamespaceDeleter) Delete(organizationID uint, clusterName string, k8
 			logger.Info("deleting kubernetes namespace")
 			err := client.CoreV1().Namespaces().Delete(ns.Name, &metav1.DeleteOptions{})
 			if err != nil {
+				// check if the namespace transitioned into 'Terminating' phase, if so
+				// ignore the error
+				namespace, errGet := client.CoreV1().Namespaces().Get(ns.Name, metav1.GetOptions{})
+				if errGet != nil {
+					return emperror.Wrapf(err, "failed to get %q namespace details", ns.Name)
+				}
+
+				if namespace.Status.Phase == corev1.NamespaceTerminating {
```

Comment:
I would negate the condition and return with the error inside the `if` instead of the `continue`

Reply:
If the state of the namespace is 'Terminating' than we can ignore the error it has no sense to retry the delete namespace operation. When the namespace deletion returns with `The system is ensuring all content is removed from this namespace. Upon completion, this namespace will automatically be purged by the system.` error the namespace gets into 'Terminating' state and eventually Kubernetes will delete it.